### PR TITLE
Create goal page added

### PR DESCRIPTION
**Pull request by sahsisunny**
# Create goal page added

# Issue #282

### What is the change?

- New folder created named `goal`.
- New files created in goal folder named `index.html`, `style.css` and `script.js`.
- Modified the main `index.html` for adding the Button to create goal. and also change button to anchor tag for better accessibility.
- Also add some css in main `style.css` for the anchor tag.

### Is Development Tested?

- [x] Yes

#### Before
Before

#### After
https://user-images.githubusercontent.com/70854507/212496454-0108b4b2-896e-492c-949c-c6af06b0ef3f.mp4

### CODE REVIEW

**Review by Maheima — DISMISSED** [review]
LGTM!

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
-node_modules/
\ No newline at end of file
+node_modules/
+package-lock.json
+createGoalTest.js
```
Comment (prakashchoudhary07) [inline .gitignore:3]: Why add package-lock.json in gitignore and why test file too?

File: index.html [hunk 2]
```diff
@@ -23,6 +24,13 @@
       >
         Create Tasks
       </button>
+      <button
+        class="profile-task-btn"
+        type="button"
+        onclick="location.href='goal/index.html'"
+      >
+        Create Goals
+      </button>
```
Comment (prakashchoudhary07) [inline index.html:33]: Why use a button to display links?

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,884 @@
...
+      }
+    },
+    "node_modules/@types/node": {
+      "version": "17.0.20",
+      "resolved": "https://registry.npmjs.org/@types/node/-/node-17.0.20.tgz",
+      "integrity": "sha512-Q15Clj3lZSLnhVA6yKw1G7SQz46DeL9gO1TEgfK1OQGvMdQ6TUWmCeWf1QBUNkw2BDfV52i2YuYd9OF3ZwGhjw==",
+      "optional": true
+    },
+    "node_modules/@types/yauzl": {
+      "version": "2.9.2",
```
Comment (prakashchoudhary07) [inline package-lock.json:27]: Why are we changing package-lock.json, when no package is installed or changed

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (sahsisunny) [reply .gitignore:3]: Because the test file is not yet complete

File: package-lock.json (hunk 3 above)
Comment (sahsisunny) [reply package-lock.json:27]: It's automatically changed when running `npm install`

File: index.html (hunk 2 above)
Comment (sahsisunny) [reply index.html:33]: Because other links like Create Task, Profiles are in button tag

File: package-lock.json (hunk 3 above)
Comment (sahsisunny) [reply package-lock.json:27]: The package-lock.json file is automatically generated and updated when running the npm install command. This file contains a detailed, versioned record of all the packages and dependencies that were installed in the project, including the exact versions of each package that were installed. This allows for consistency and reproducibility when installing dependencies on different machines or at different times. By - ChatGPT🤨

File: index.html (hunk 2 above)
Comment (sahsisunny) [reply index.html:33]: Can you create a ticket for this issue that to navigate from one page to another there should be anchor tag and this is the best practice ?

File: .gitignore (hunk 1 above)
Comment (sahsisunny) [reply .gitignore]: reverted

File: package-lock.json (hunk 3 above)
Comment (sahsisunny) [reply package-lock.json]: reverted

File: index.html (hunk 2 above)
Comment (sahsisunny) [reply index.html]: done

File: index.html [hunk 4]
```diff
@@ -10,34 +10,33 @@
...
+        class="element-display-remove link-button"
         href="user-management/index.html"
       >
         User Management
       </a>
-      <a href="online-members/online-members.html" class="online-members-link">
+      <a
+        href="online-members/online-members.html"
+        class="online-members-link link-button"
+      >
```
Comment (prakashchoudhary07) [inline index.html:39]: I think you should make these changes in separate PR, Things that are not related to your PR are to be done in a new PR. Please raise a new PR for this

File: createGoalTest.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+const puppeteer = require('puppeteer');
+let config = {
+  launchOptions: {
+    headless: false,
+  },
+};
+puppeteer.launch(config.launchOptions).then(async (browser) => {
+  const page = await browser.newPage();
+  await page.goto('http://192.168.22.108:5500/goal/index.html');
```
Comment (prakashchoudhary07) [inline createGoalTest.js:9]: Why not URL here, and why is there an IP address here
And will this test Pass in CI or remote test environments?

File: index.html (hunk 4 above)
Comment (sahsisunny) [reply index.html]: done

File: createGoalTest.js (hunk 5 above)
Comment (sahsisunny) [reply createGoalTest.js]: done

File: index.html [hunk 6]
```diff
@@ -24,6 +25,7 @@
       >
         Create Tasks
       </button>
+      <a class="action-button" href="goal/index.html"> Create Goals </a>
       <button
         class="profile-task-btn"
         type="button"
```
Comment (Maheima) [inline index.html:31]: This looks off to me. Some elements are buttons here, some are anchors. Either add them as the other elements were added and create a separate PR to convert them all to button/anchors  or convert them all in this one itself however @prakashchoudhary07 thinks it should be done

File: goal/index.html [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+          </select>
+        </div>
+        <div class="inputBox">
+          <label for="type" class="editable"
+            >Type:<span class="required">*</span></label
+          >
+          <select required class="input notEditing" id="type" name="type">
+            <option disabled selected>Select a type</option>
+            <option value="FEATURE">Feature</option>
+            <option value="BUG">Bug</option>
```
Comment (Maheima) [inline goal/index.html:106]: adding `id` to tags which will share the same styling (here priority, type, user-role) is not a good idea. Use class instead. That way you'll have just one class name that you can add to all 3 of them

**Review by Maheima — CHANGES_REQUESTED** [review]

File: goal/index.html (hunk 7 above)
Comment (sahsisunny) [reply goal/index.html:96]: changed id to class in some elements

File: index.html (hunk 6 above)
Comment (sahsisunny) [reply index.html]: changed

**Review by prakashchoudhary07 — APPROVED** [review]
